src/components/variant/ui/drawer-window.tsx

```diff
@@ -0,0 +1,230 @@
+                        checked={filterSelection !== normClass}
+                        className="ml-1"
+                        onChange={(e: any) => handleSelection(e.target.checked)}
+                      />
+                    )}
+                  </td>
+                ))}
+              </tr>
+            </thead>
+          )}
+
+          <tbody>
+            {rows?.map((row, index) => {
+              if (!row) return <tr key={index} />
+
+              const blueBg = ' p-3 bg-blue-darkHover'
+              return (
+                <tr key={row.name}>
+                  <Tooltip
+                    overlay={row.tooltip}
+                    placement="bottomLeft"
+                    trigger={row.tooltip ? ['hover'] : []}
+                  >
+                    <td
+                      className={cn(
+                        'p-3 text-blue-bright whitespace-nowrap sticky left-0',
+                        `${shouldAddShadow ? blueBg : ''}`,
+                      )}
+                    >
+                      <span
+                        className="cursor-auto"
+                        onMouseDownCapture={onMouseDownHandler}
+                      >
+                        {row.title}
+                      </span>
+                    </td>
+                  </Tooltip>
+
+                  {row.cells
+                    .filter(cell => cell[1]?.includes(filterSelection))
+                    .map((cell, cIndex) => (
+                      <td
+                        key={cIndex}
+                        className={cn(
+                          'py-3 pr-3 font-normal',
+                          cell[0].includes('</a>')
+                            ? 'text-blue-bright'
+                            : !cell[1]?.includes(noTrHitClass) &&
+                                'text-grey-blue',
+                        )}
+                      >
+                        <span
+                          className="cursor-auto"
+                          onMouseDownCapture={onMouseDownHandler}
+                        >
+                          {cell[0]}
+                        </span>
+                      </td>
+                    ))}
+                </tr>
+              )
+            })}
+          </tbody>
+        </table>
+      )}
+    </div>
+  )
+}
+
+export const DrawerWindow = observer(
+  ({ aspect, layout }: { aspect: TRecCntResponse; layout: IGridLayout[] }) => {
+    const ref = useRef<HTMLDivElement>(null)
+
+    const [startedLeftDistance, setStartedLeftDistance] = useState<
+      number | null
+    >(null)
+
+    const [shouldAddShadow, setShouldAddShadow] = useState(false)
+
+    const getLeftDistance = (element: HTMLDivElement | null): number | null => {
+      const tableNode = element?.children?.[0]?.children?.[0]
+      const tbodyNode = tableNode?.children[tableNode?.children.length - 1]
+      const trackedTdNode = tbodyNode?.children?.[0]?.children?.[1]
+
+      if (!trackedTdNode) return null
```

Comment:
why necessarily return null? You can return `trackedTdNode?.getBoundingClientRect().left` which would be undefined